### PR TITLE
fix custom meta DB remover

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
## Reasoning behind the pull request
- the fix done in #4520 was not enabled for metachain as well 
  
## Proposed changes
- use the factory for custom db remover

## Testing procedure
- same as #4520 , but for meta nodes only

## Pre-requisites

Based on the [Contributing Guidelines](CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by andreibancioiu — DISMISSED** [review]

File: storage/factory/pruningStorerFactory.go [hunk 1]
```diff
@@ -301,6 +300,9 @@ func (psf *StorageServiceFactory) CreateForShard() (dataRetriever.StorageService
 	return store, err
 }
 
+// TODO: split in 2 components shardStorageServiceCreator and metStorageServiceCreator that have a base which will contain the
```
Comment (SebastianMarian) [inline storage/factory/pruningStorerFactory.go:303]: typo: metStorageServiceCreator

**Review by SebastianMarian — APPROVED** [review]

File: storage/factory/pruningStorerFactory.go (hunk 1 above)
Comment (iulianpascalau) [reply storage/factory/pruningStorerFactory.go:303]: I've expected the proper refactor work

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

**Review by iulianpascalau — APPROVED** [review]
Green just to unlock this PR since I've pushed in the branch

**Review by SebastianMarian — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**